src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -48,56 +49,76 @@ public CodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringServic
             var result = new ArrayBuilder<object>();
             foreach (var codeAction in codeActions)
             {
+                if (!clientSupportsWorkspaceEdits)
```

Comment:
I think this check could potentially be moved outside the loop?  Since the clientSupportsWorkspaceEdits is valid for the entire request, not just a particular code action

Reply:
I still need to iterate over the `codeActions` though, right? If I move the check out, then I have to make (and maintain) two separate loops over `codeActions`...

Reply:
ah true, and you actually need to do AddCodeActionWithoutIncludingChanges in the second loop as well as including changes.  yup I see